src/scripts/js-telephone-input.js

```diff
@@ -0,0 +1,487 @@
+
+    //----------------------------------------------------
+    // Define our constructor
+    //----------------------------------------------------
+    global.jsTelephoneInput = function() {
+
+        //Check if parameters exists.
+        if(!arguments[0] || !arguments[1] || typeof arguments[1] !== "object") {
+            console.log('js-telephone-input: field or parameters are not defined');
+            return;
+        }
+
+        // Create global elements references
+        this.LibPhoneNumber = require('awesome-phonenumber');
+        this.countriesList = require('./country-list');
+        this.field = arguments[0];
+        this.classNames = {
+            tel : 'ui-telephone',
+            telError : ' ui-telephone--error',
+            telSuccess : ' ui-telephone--success',
+            telWithFlag : ' ui-telephone--with-flag',
+            telCanChangeFlag : ' ui-telephone--change-flag',
+            telNumber : 'ui-telephone__number',
+            telFlags : 'ui-telephone__flags',
+            telFlagsSelected : 'ui-telephone__selected-flag',
+            telFlagsList : 'ui-telephone__flags-list',
+            telFlagsListItem : 'ui-telephone__flags-list-item',
+            telFlagsListItemName : 'ui-telephone__flag-name',
+            telFlagsListItemIcon : 'ui-telephone__flag-icon',
+            telFlagsListItemCode : 'ui-telephone__flag-code',
+            validations: 'ui-telephone__validations',
+            validationError : 'ui-telephone__validation--error',
+            validationInfo : 'ui-telephone__validation--info',
+            validationSuccess : 'ui-telephone__validation--success',
+            inputErrorClass : ' ui-telephone__input-error',
+            showClass : ' is-visible'
+        };
+        this.parameters = extendDefaults(arguments[1], this.field);
+        this.validations = getValidations(this);
+        this.elements = getElements(this);
+        this.preventKeyup = false;
+        this.phoneNumber = null;
+        this.exampleNumber = null;
+
+        //Validate User Default Value
+        validateInitialValue(this);
+
+        //Define field events and attributes
+        addEvent(this.field, 'keyup', format, this);
+        addEvent(this.field, 'click', format, this);
+        addEvent(this.field, 'keypress', checkPressedKey, this);
+        addEvent(this.field, 'blur', this.setInputsValue, this);
+        setExampleNumber(this.parameters.country, this);
+    };
+
+    //----------------------------------------------------
+    // Public Methods
+    //----------------------------------------------------
+
+    // Sets the value on the hidden inputs
+    jsTelephoneInput.prototype.setInputsValue = function(event, component) {
+        if(this instanceof global.component) {
+            component = this;
+        }
+        var valid = component.isValid(),
+            telephone = component.getTelephone(),
+            numberInput = document.getElementById(component.parameters.number),
```

Comment:
Considerando que [`querySelector` y `querySelectorAll` ya tienen un soporte del 97% global](http://caniuse.com/#feat=queryselector)] (IE >= 9), para no depender de los selectores por `id` se podría hacer que los componentes de la UI usen el atributo `data-js` mencionado anteriormente y reemplazar las llamadas de `getElementById` por las siguientes:

```js
// assuming: this.field instanceof HTMLElement === true
var numberInput = this.field.querySelector('[data-js="ui-telephone__number-input"]');
```

Se puede usar `querySelector` desde `this.field` o `component.field` para limitar la búsqueda en el DOM a los `childNodes` del root element del componente.